core/domain/user_jobs_continuous.py

```diff
@@ -428,15 +428,28 @@ def reduce(key, stringified_values):
         # Find the average of all average ratings
         ratings = [value['average_rating_for_owned_exp'] for value in values
                    if value.get('average_rating_for_owned_exp')]
+        mr_model = user_models.UserStatsModel.get(key, strict=False)
+
         if len(ratings) != 0:
             average_ratings = (sum(ratings) / float(len(ratings)))
-            user_models.UserStatsModel(
-                id=key,
-                impact_score=user_impact_score,
-                total_plays=total_plays,
-                average_ratings=average_ratings).put()
         else:
-            user_models.UserStatsModel(
-                id=key,
-                impact_score=user_impact_score,
-                total_plays=total_plays).put()
+            average_ratings = None
+
+        if not mr_model:
+            if average_ratings:
+                user_models.UserStatsModel(
+                    id=key,
+                    impact_score=user_impact_score,
+                    total_plays=total_plays,
+                    average_ratings=average_ratings).put()
+            else:
+                user_models.UserStatsModel(
+                    id=key,
+                    impact_score=user_impact_score,
+                    total_plays=total_plays).put()
+        else:
+            mr_model.impact_score = user_impact_score
+            mr_model.total_plays = total_plays
+            if average_ratings:
```

Comment:
Removed this check. 
@wxyxinyu - Average ratings is not set 0 anywhere, it's always set to be None.